homeassistant/components/zwave_js/climate.py

```diff
@@ -193,6 +193,16 @@ def __init__(
         self._set_modes_and_presets()
         if self._current_mode and len(self._hvac_presets) > 1:
             self._attr_supported_features |= ClimateEntityFeature.PRESET_MODE
+        if HVACMode.OFF in self._hvac_modes:
+            self._attr_supported_features |= ClimateEntityFeature.TURN_OFF
+
+            # We can only support turn on if we are able to turn the device off,
+            # otherwise the device can be considered always on
+            if len(self._hvac_modes) == 2 or any(
```

Comment:
Don't need the any test as testing with len>=2 should be enough (as we know `off` is one of them).

Reply:
some HVAC units support `dry` and `fan_only`. The default async_turn_on doesn't handle those modes if e.g. the hvac modes are [off, dry, fan_only], the turn on service won't do anything because there are more than 2 but they are not heat, cool, or heat_cool. I don't think we want to expose the turn on service in that case

Reply:
I mean we only get to this point if `off` is already part of `hvac_modes` so in your example it would add neither turn_on or turn_off

Reply:
Right and in my example off is a part of the list 🙂

Reply:
I think we should as if we can turn it off shouldn't we be able to turn it on?

Didn't look at closely at `climate` but you mean it doesn't handle turn_on for fan_only (as example)?

Reply:
That's correct but you are also correct that turn on and turn off go hand in hand, wouldn't make sense having just one. I will adjust the logic slightly 👍

Reply:
> I think we should as if we can turn it off shouldn't we be able to turn it on?
Then why are there two seperate capabilities?

I thought the point of adding these supported features is because `turn_on` is ambigous for some devices. However `turn_off` isnt if it has a `off` hvac_mode. I agree with @raman325's point of view.

Reply:
Also a good point.

Reply:
OK so I see this as the right implementation to avoid a breaking change. The reason is that today, you can have a turn_on function that does nothing and a turn_off function that turns the device off in the scenario I described above. So it makes sense that the logic is implemented this way.

I will reevaluate the logic in the other PR because in that we **are** making a breaking change by changing the behavior of `turn_on` to accommodate the extra modes

Reply:
For your own context @ramas, I maintain a climate integration that actually does let me issue a turn on command without a HVAC mode and it resumes whatever HVAC mode was last used independent of Home Assistant, so for me this feature is great because I actually do support both turn on and turn off, and turn on is actually different than changing HVAC modes.

Reply:
Nice wish the zwave standard supported that. Although now I'm wondering, it may be possible. Will have to test and adjust the other PR accordingly if so